core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPostProcessingB.kt

```diff
@@ -0,0 +1,202 @@
+package fr.sncf.osrd.stdcm.graph
+
+import fr.sncf.osrd.api.pathfinding.makeOperationalPoints
+import fr.sncf.osrd.envelope_sim.TrainPhysicsIntegrator
+import fr.sncf.osrd.envelope_sim.allowances.utils.AllowanceValue
+import fr.sncf.osrd.envelope_sim_infra.EnvelopeTrainPath
+import fr.sncf.osrd.graph.Pathfinding
+import fr.sncf.osrd.graph.Pathfinding.EdgeLocation
+import fr.sncf.osrd.graph.Pathfinding.EdgeRange
+import fr.sncf.osrd.graph.PathfindingEdgeLocationId
+import fr.sncf.osrd.graph.PathfindingEdgeRangeId
+import fr.sncf.osrd.sim_infra.api.*
+import fr.sncf.osrd.stdcm.STDCMResult
+import fr.sncf.osrd.stdcm.preprocessing.interfaces.BlockAvailabilityInterface
+import fr.sncf.osrd.train.RollingStock
+import fr.sncf.osrd.train.RollingStock.Comfort
+import fr.sncf.osrd.train.TrainStop
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.meters
+import java.util.*
+
+/**
+ * This class contains all the static methods used to turn the raw pathfinding result into a full
+ * response. This includes creating the final envelope (merging the parts + applying the allowances)
+ */
+class STDCMPostProcessingB(private val graph: STDCMGraph) {
+    /**
+     * Builds the STDCM result object from the raw pathfinding result. This is the only non-private
+     * method of this class, the rest is implementation detail.
+     */
+    fun makeResult(
+        infra: RawSignalingInfra?,
+        path: Result,
+        startTime: Double,
+        standardAllowance: AllowanceValue?,
+        rollingStock: RollingStock?,
+        timeStep: Double,
+        comfort: Comfort?,
+        maxRunTime: Double,
+        blockAvailability: BlockAvailabilityInterface
+    ): STDCMResult? {
+        val ranges = makeEdgeRange(path)
```

Comment:
I don't think we need edge *ranges* anymore here, we always use STDCM edges in full. There's enough info inside to project them on blocks

edit: I assume it's to keep the rest of the file mostly unchanged? It makes sense if it's not entirely stable yet